File: neetcode/Data Structures & Algorithms/duplicate-integer/submission-2.rs
impl Solution {
    pub fn has_duplicate(mut nums: Vec<i32>) -> bool {
        nums.sort_unstable();
        for i in 1..nums.len() {
            if nums[i] == nums[i - 1] {
                return true;
            }
        }
        
        false
    }
}

File: neetcode/check-if-array-is-sorted-and-rotated/submission-2.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        found = 0
        i = 0
        for i in range(n):
            if nums[i] > nums[(i + n + 1) % n]:
                found += 1

        return found == 0 or found == 1
        

        